content/oci/oci.go

```diff
@@ -190,9 +190,8 @@ func (s *Store) Delete(ctx context.Context, target ocispec.Descriptor) error {
 		}
 		if s.AutoGC {
 			for _, d := range danglings {
-				// do not delete existing manifests in tagResolver
-				_, err = s.tagResolver.Resolve(ctx, string(d.Digest))
-				if errors.Is(err, errdef.ErrNotFound) {
+				// do not delete existing tagged manifests
+				if !s.isTagged(d) {
```

Comment:
Since we now automatically delete untagged successor manifests when `Store.AutoGC` is enabled, we can further automatically delete referrers as well. That is, we no longer need the `s.AutoDeleteReferrers` option. 
Instead, enabling `Store.AutoGC` will mean:

- Automatically delete untagged successor manifests while keeping tagged successor manifests
- Automatically delete referrers, no mater it is tagged or not

If we agree on this, we will need another PR to make this change. /cc @shizhMSFT

Reply:
See #683